app/instance-initializers/bugsnag.js

```diff
@@ -1,38 +1,80 @@
 import Ember  from 'ember';
 import config from '../config/environment';
 import { getContext } from 'ember-cli-bugsnag/utils/errors';
-import { getMetaData } from '../utils/bugsnag';
+import { getMetaData, getUser } from '../utils/bugsnag';
 import Bugsnag from 'bugsnag';
 
+const {
+  get,
+  setProperties,
+} = Ember;
+
+export function initialize(instance) {
+  if (Bugsnag.apiKey === undefined) {
+    return;
+  }
+  const currentEnv = config.environment;
+  const bugsnagConfig = config.bugsnag || {};
+  const releaseStage = bugsnagConfig.releaseStage || currentEnv;
+
+  if (currentEnv !== 'test' && Bugsnag.notifyReleaseStages.indexOf(releaseStage) !== -1) {
+    const owner = instance.lookup ? instance : instance.container;
+    const router = owner.lookup('router:main');
+
+    setProperties(this, {
+      owner,
+      router,
+    });
+
+    Ember.onerror = (error) => this._onError(error);
+
+    router.didTransition = this._didTransition();
+  }
+}
+
 export default {
   name: 'bugsnag-error-service',
 
-  initialize: function(instance) {
+  initialize,
+
+  _didTransition() {
+    const router = get(this, 'router');
+    const originalDidTransition = router.didTransition || function() {};
+
+    return function() {
+      Bugsnag.refresh();
+      return originalDidTransition.apply(this, arguments);
+    };
+  },
+
+  _onError(error) {
+    this._setContext();
+    this._setNotifyException(error);
+    this._setUser();
 
-    if (Bugsnag.apiKey === undefined) {
-      return;
+    /* eslint-disable no-console */
+    console.error(error.stack);
+    /* eslint-enable no-console */
+  },
+
+  _setContext() {
+    const router = get(this, 'router');
+    Bugsnag.context = getContext(router);
+  },
+
+  _setNotifyException(error) {
+    const owner = get(this, 'owner');
+    if (getMetaData) {
```

Comment:
I think this will stop the error from being reported if `getMetaData` didn't get setup correctly. We need to avoid a situation where consumer's errors go silently unreported so I'd like the error to still go through.